diff --git a/public/scripts/world-info.js b/public/scripts/world-info.js
index 621d6d3..bdf5dbb 100644
--- a/public/scripts/world-info.js
+++ b/public/scripts/world-info.js
@@ -6059,6 +6059,27 @@ function updateAuxBooks(fileName, computeNext) {
     saveSettingsDebounced();
 }
 
+function initializeGlobalWorldInfoSelector() {
+    $('#world_info').select2({
+        width: '100%',
+        placeholder: t`No Worlds active. Click here to select.`,
+        allowClear: true,
+        closeOnSelect: false,
+    });
+
+    select2ChoiceClickSubscribe($('#world_info'), target => {
+        const name = $(target).text();
+        const selectedIndex = world_names.indexOf(name);
+        const alreadySelectedInEditor = $('#world_editor_select option:selected').text() === name;
+        if (selectedIndex !== -1 && !alreadySelectedInEditor) {
+            $('#world_editor_select').val(selectedIndex).trigger('change');
+            console.log('Quick selection of world', name);
+        } else {
+            console.warn('lets not reload an already loaded list yes?');
+        }
+    }, { buttonStyle: true, closeDrawer: true });
+}
+
 export function initWorldInfo() {
     $('#world_info').on('mousedown change', async function (e) {
         // If there's no world names, don't do anything
@@ -6261,28 +6282,10 @@ export function initWorldInfo() {
             closeOnSelect: true,
             multiple: false,
         });
-
-        $('#world_info').select2({
-            width: '100%',
-            placeholder: t`No Worlds active. Click here to select.`,
-            allowClear: true,
-            closeOnSelect: false,
-        });
-
-        // Subscribe world loading to the select2 multiselect items (We need to target the specific select2 control)
-        select2ChoiceClickSubscribe($('#world_info'), target => {
-            const name = $(target).text();
-            const selectedIndex = world_names.indexOf(name);
-            const alreadySelectedInEditor = $('#world_editor_select option:selected').text() === name;
-            if (selectedIndex !== -1 && !alreadySelectedInEditor) {
-                $('#world_editor_select').val(selectedIndex).trigger('change');
-                console.log('Quick selection of world', name);
-            } else {
-                console.warn('lets not reload an already loaded list yes?');
-            }
-        }, { buttonStyle: true, closeDrawer: true });
     }
 
+    initializeGlobalWorldInfoSelector();
+
     $('#WorldInfo').on('scroll', () => {
         $('.world_entry input[name="group"], .world_entry input[name="automationId"]').each((_, el) => {
             const instance = $(el).autocomplete('instance');
